### ISSUE TITLE
The Data Science Tree of Knowledge

### ISSUE THREAD

**Issue by guevara** [guevara/read-it-later#7322]
The Data Science Tree of Knowledge<br>
<br>
https://ift.tt/35YrejE<br>
<br>
About 0 Minutes<br>
<br>
<div>
<div>
<section>
<article>
<h2>The Data Science Tree of Knowledge - What is Data Science and How to Educate Data Scientists</h2>
<time datetime="2020-07-07T00:00:00+00:00">07 Jul 2020</time>
<p><strong>TL;DR</strong></p>
<p>Here it is:</p>
<p><img src="https://pabloinsente.github.io/../assets/post-14/ds-tree.svg"></p>
<p>Now the essay.</p>
<h2>Data Science challenges</h2>
<p>As with any other science-based discipline, Data Science tools and core of knowledge changes constantly. During my sociology, public policy, and psychology studies, I quickly learned that research perpetually redefines the boundaries of knowledge. Through friends in fields like physics, biology, engineering, and medical sciences, I have learned that the situation is no different there. Yet, I believe there are a couple of characteristics that make Data Science a particularly hard to &ldquo;keep up with&rdquo; as a field. More than most fields I am aware of at least. If you are active in social media, you will quickly find testimonies of people feeling overwhelmed by the pace of change of the field, and its overall fuzziness. With this in mind, in this brief essay, I try to answer the following questions: (1) <strong>What is Data Science?</strong>; (2) <strong>Why is so hard to &ldquo;keep up with&rdquo; Data Science tools and knowledge?</strong>; (2) <strong>What can we do about it (question 2)?</strong></p>
<h2>What is Data Science? Buzzword or discipline?</h2>
<p>There is no universal agreement around what Data Science &ldquo;is&rdquo;. True, you may have read or heard things like &ldquo;<em>Data Science is the intersection of computer science, mathematics/stats, and science</em>&rdquo; or maybe &ldquo;<em>Data Science is the application of data-related tools and knowledge to solve practical problems</em>&rdquo;. However, there is no reason for you to accept such definitions. There is nothing special about them. All of them are essentially <em>proposals</em>, no matter how many Twitter followers, degrees, academic papers, or awards the definition proponent possesses. Based on this fact, you may contest the whole &ldquo;<em>What Data Science is</em>&rdquo; as a pseudo-problem. Maybe Data Science is just another empty label created by people who wanted to profit from making applied statistics to sound cooler. If I had a dollar for every time I read/heard Data Science is just &ldquo;applied stats&rdquo; I would be doing something fun my pile of money instead of writing this.</p>
<p>But this is the thing: <strong>the whole &ldquo;what a discipline really is&rdquo; is a universal problem</strong>. If you have had a chance to read about the history and epistemological foundations of any discipline, you will find that more often than not, multiple definitions and perspectives about what a discipline &ldquo;really is&rdquo;, each one with its group of advocates and followers. Just go to ask a cognitive psychologist and a behavioral psychologist for a definition of psychology. You will end up with 4 or 5 definitions (yeah, from just 2 people). So, the whole &ldquo;Platonic&rdquo; conundrum about what disciplines &ldquo;really are&rdquo; should not deter you from taking a discipline seriously. Particularly with disciplines that can impact large numbers of individuals, especially vulnerable individuals, as Data Science. In any case, whatever you think Data Science &ldquo;really is&rdquo;, probably has something to do with things like data, mathematics, algorithms, and human insight, blended in such a way that makes sense.</p>
<p>&ldquo;<em>Got it. There are many reasonable definitions of Data Science and it may be a thing. So what?</em>&rdquo; You see, I do care about this because your understanding of what Data Science &ldquo;is&rdquo;, provides substance to the discussion of why, in my view, it is so damn hard to keep up with the pace at which things change. Let me begin by claiming the following: Data Science is an applied discipline, like engineering or medicine. This may or may not amount to anything to you but saying that it is not like physics or pure mathematics is a step forward in my view. Take applied fields like engineering and medicine. They exist in a duality: they have their <strong>own goals and points of view</strong>, but at the same time are <strong>constrained by</strong> the evolution of disciplines like physics, biology, mathematics, and computer science (among others). It is pretty transparent that Data Science also depends on the evolution of computer science and mathematics, but does Data Science have any &ldquo;independent&rdquo; goal? Any &ldquo;unique&rdquo; point of view? Its unique concerns? If Data Science cannot find any particular &ldquo;angle&rdquo; and unique &ldquo;space of concerns&rdquo;, there is little point to raise Data Science to the category of &ldquo;Discipline&rdquo;.</p>
<p>With the aforementioned ideas in mind, let me define Data Science goals as follow: answering user-defined meaningful questions by applying scientific reasoning and mathematical and computational tools. Great, but what kind of questions? Electrical engineering is about, well, electricity. Hence, electrical engineering deals with questions related to the technology of electricity. For instance, does a question about women&rsquo;s health which applies the scientific method and mathematical and computational tools qualify as a &ldquo;Data Science question&rdquo;? Probably yes, but it also qualifies as a Public Health question, and Public Health experts exist already, so why on earth to invent a new label for something it has one already?</p>
<p>This is the main challenge for people who self-identify as Data Scientist: <strong>what makes you different from an econometrician, or a biostatistician, or a demographer, or a computer scientist, and any other data expert within a field?</strong> As it now, it sounds like we are just playing a &ldquo;labels game&rdquo; for convenience (marketability). I have read/heard people pointing out to their programming skills &ldquo;<em>I am a better programmer than any biostatistician I know of</em>&rdquo;. If this answer suffices for you, cool, you can skip the rest of this article. Otherwise, if you are like me, such an answer is unsatisfactory. After all, neither biostatistics nor Data Science is a programming-skills competitions, and what counts as a &ldquo;good programmer&rdquo; is not clear at all.</p>
<p>In my view, what sets apart and helps to Define Data Science as a discipline, it is the how-to jointly use data, mathematics, and computational tools, to address empirical questions IN ITSELF. If this sounds like a reiteration of my first definition, pause and ponder for a second. It is not. The key is the &ldquo;IN ITSELF&rdquo; part of it. Rephrasing it, is about the art and science of the question-answering with data plus math plus computational tools. &ldquo;Wait, isn&rsquo;t that the goal of computer science or statistics&rdquo;. It is not. At least not necessarily. Strictly speaking, statistics can generate new and useful knowledge without any regard for, for instance, practical computational constraints. True, ultimately, to be useful, you must pay attention to things like data collection procedures and computational resources, but the point is that this is not required in principle. Same with computer science, which is the study of computation and information. How you get such data, or how to formulate appropriate questions, or how to apply the scientific method, are not concerns of computer science, not in principle at least. Yet, I propose, they must be concerns of Data Science and Data Scientist, always.</p>
<p>What about biostatistics? Or Econometrics? After all, it sounds like they do care about all that stuff at once. Here is my issue: econometricians do not care (no in principle) whether the methods will be useful for biology, and biostatistics do not care whether their methods are appropriated for microeconomics applications. More generally, they do not care about developing tools and methods to answer empirical questions <strong>in general</strong>, and tend to be a bit indifferent to things like computer architectures and other disciplines&rsquo; algorithms. <strong>But those issues MUST be of consideration for Data Science as a discipline</strong> (as I understand it). Ideally, Data Science should be able, as a discipline, to address the mathematical and computational needs of a wide variety of disciplines and fields of inquiry. &ldquo;But NOBODY can do that, Pablo!&rdquo; Of course, no single individual can! In the same manner that no single medical student can address all medical problems of the human species. But medicine, as a discipline, aims to address that issue, and Data Science as a discipline should have such a general scope. Here are some examples of things that a Data Scientist should care about:</p>
<ul>
<li>What is the best way to store data from different sizes and formats?</li>
<li>When to automate data processing pipelines?</li>
<li>When is it a good idea to use a declarative versus an imperative approach to data visualization?</li>
<li>What questions are appropriated to ask when using unsupervised learning methods?</li>
<li>At what point is it a good idea to move your data processing pipeline to a cluster of computers in the cloud?</li>
<li>In what circumstances quasi-experimental methods are an acceptable replacement for randomized experimental methods?</li>
<li>At what point is worth writing code in low-level programming languages like C++ or Fortran rather than Python or R?</li>
</ul>
<p>These are just a couple of examples of questions which, I believe, share the two traits: <strong>generality</strong>, i.e., are relevant for a wide variety of fields of expertise, and cross-discipline, meaning that they require knowledge and skills of several fields to be answered.</p>
<p>With this idea in mind, and assuming you find my perspective acceptable, I can address the next question.</p>
<p>Short answer: because of its generality and immaturity. Trying to keep up with &ldquo;medical sciences&rdquo;, as a whole, is impossible for any finite human being. Keeping up with Data Science as a discipline, it is equally impossible. I believe many people in the field feel overwhelmed because they are constantly bombarded with the latest and greatest advances in &ldquo;Data Science&rdquo;, which can come from: time-series analysis, declarative data visualization, computer vision, affective computing, natural language processing, emotion synthesis, econometrics, biostatistics, signal processing, applied mathematics, information theory, new python libraries, new R libraries, new MATLAB libraries, new JS libraries, new computer architectures, new GPUs, new Big Data technologies, new optimization techniques, and the list goes on and on. <strong>The broad scope of Data Science is a course for its practitioners</strong>.</p>
<p>One of the reasons sometimes it feels like Data Science is impossible to narrow down in scope, is because of hiring practices. For example, I have gone through at least 20 cycles of interview processes, from places as diverse as Airbnb, Asana, Trivago, McKinsey &amp; Company, and Akuna Capital, and pretty much all of them were different. &ldquo;<em>Did you even look at the job description?</em>&rdquo; The hell I did, and no, the job description was of little use to narrow down my preparation, as most of them incorporated all skills and technologies under the sun. I have been asked to solve problems using traditional machine learning techniques, using deep learning techniques, to implementing advance SQL queries, to create engaging and insightful graphs, to design randomized controlled experiments, to utilize quasi-experimental econometric techniques, to solve complex algorithm problems, to code in R, to code in Python, and the list goes on and on. When the market implicitly defines Data Science as the whole field at once, you are doom to feel overwhelmed by the share size and pace of change of the field. You are never ready, you are never good enough, you will never master your craft because the craft is too big and brittle to master.</p>
<p>Another reason is the eagerness from academia, industry, government, and even single contributors, to innovate and &ldquo;put things out there&rdquo; as fast as they can. Not that constant investment and innovation is bad, not at all, but it does create a perpetual cycle of fast obsolescence of skill and knowledge. &ldquo;<em>Oh, so you spent a year becoming a TensorFlow 1.X expert? Too bad, we want TensorFlow 2.X now</em>&rdquo; Or &ldquo;<em>Wait, no, we changed our mind, we want PyTorch experts now. But you know what, we learned bout this great new thing, MXNet, we heard that Amazon use it, and we want to follow their lead, so we want experts in both</em>&rdquo;. Or changing from R to Python. Or mastering both natural language processing AND time-series analysis, because, why not, how hard it can be, right?</p>
<p>Last but not least, I mentioned Data Science immaturity. This implies two things: (1) most people have no idea of where does Data Science begins and ends, or what a Data Scientists is supposed to look like, or what they can or cannot do; (2) most people have no idea how to educate Data Scientists, or what does constitute a well-stablished core of knowledge for the discipline.</p>
<p>Engineering and medicine are old enough disciplines that there is some degree of agreement regarding what they are, how do they look like, and what are they expected roles and preparation. Most schools more or less agree on how to educate engineers and physicians, and a <strong>sensible core of knowledge</strong> has been defined as a basis for the discipline. I believe, this is what Data Science is missing: an agreement about the identity, roles, preparation, and what a <strong>sensitive core of knowledge</strong> looks like.</p>
<h2>What to do about it</h2>
<p>In my experience, when I point out an issue, people are quick to ask me for a solution, usually a bit annoyed by my impertinence. I do not know how to fix the issue as a whole, but I do have a couple of ideas, particularly on the side of Data Science education and definition of a &ldquo;sensible core of knowledge&rdquo;. This I what I - presumptuously- call &ldquo;<strong>The Data Science Tree of Knowledge</strong>&rdquo;.</p>
<h2>The Data Science Tree of Knowledge</h2>
<p>Before offering my perspective about what should constitute the &ldquo;Data Science Tree of Knowledge&rdquo;, I want to briefly look at what &ldquo;top&rdquo; programs in the world are offering as an educational core at the undergraduate and graduate level in Data Science.</p>
<p>UC Berkeley offers a good example of the disciplinary basis for Data Science education:</p>
<div>
<div>
<pre><code>- Foundations of Data Science
- Calculus I and II
- Linear Algebra
- Program Structures
- Data Structures
- Domain Emphasis
</code></pre>
</div>
</div>
<p>At the graduate level, the NYU PhD in Data Science is a great example:</p>
<div>
<div>
<pre><code>- Introduction to Data Science
- Probability and Statistics for Data Science
- Machine Learning
- Big Data
- Inference and Representation
</code></pre>
</div>
</div>
<p>Another way to assess this is by looking at eligibility requirements for graduate programs. The University of Toronto&rsquo;s MS in Data Science specifics:</p>
<div>
<div>
<pre><code>    At least two senior level statistics courses equivalent to:
        STA 302H Applied Regression Analysis, and/or
        STA 347H Probability Theory, and/or
        STA 452H/453H Introduction to Mathematical Statistics

    At least three senior level computer science courses equivalent to:
        CSC 373H Algorithms and complexity
        CSC 343H Database Systems
        CSC 369H Operating Systems
</code></pre>
</div>
</div>
<p>Undergraduate and graduate education operate under time, financial, and human capital constraints, so it is expected they will not put everything they would like to in the curriculum.</p>
<p>My goal is a bit more abstract and unconstrained by the factual realities of universities and professional degrees. I want to propose an <strong>idealized yet sensible corpus of knowledge for the discipline</strong>, which may serve as a foundation for Data Science education. I use the <strong>Tree of Knowledge metaphor</strong>, as it nicely represents the interconnectedness, fuzziness, and importance of strong foundations I want to convey.</p>
<h3>Roots</h3>
<p>The roots of the tree are interconnected disciplines which provide the foundation&rsquo;s Data Science as a discipline and Data Scientists as practitioners. In an ideal world, these are the subject people should spend most of their time and effort, as strong foundations will allow for quicker and, more importantly, higher-quality learning of subsequent subjects. In sum, these are the topics I think all Data Scientists should learn about regardless of their future area of expertise:</p>
<ul>
<li>Calculus</li>
<li>Linear algebra</li>
<li>Probability theory and statistics</li>
<li>Discrete mathematics (set theory)</li>
<li>Intro to optimization</li>
<li>Intro to information theory</li>
<li>Intro to algorithms</li>
<li>Intro to programming</li>
<li>Intro to computer science</li>
<li>Intro to data structures</li>
<li>Intro to data management</li>
<li>Scientific method and reasoning</li>
<li>Science communication</li>
<li>Quantitative and qualitative research methods</li>
<li>Philosophy of science</li>
<li>Intro to data visualization</li>
</ul>
<h3>Trunk</h3>
<p>The trunk of the tree represents a collection of topics that can be selected depending on interest and area of desired expertise. Not all Data Scientists need to learn about Deep Learning. If you have been following all the buzz about Deep Learning, you may think this is utter nonsense. &ldquo;EVERYONE should know about Deep Learning&rdquo;. Well, I do not believe so. Becoming a proficient Deep Learning practitioner takes an enormous amount of time and dedication. I do not believe you can master experimental and quasi-experimental methods, AND Deep Learning at the same time. True, it is useful to know what is &ldquo;Deep Learning all about&rdquo; but that it is completely different from becoming a Deep Learning practitioner. So, it is nice to have notions about most subjects, but practitioners should focus on a synergistic (is that even a word?) and a manageable subset of topics like:</p>
<ul>
<li>Advance optimization</li>
<li>Intermediate algorithms</li>
<li>Machine learning foundations</li>
<li>Deep learning foundations (neural networks)</li>
<li>Applied statistics</li>
<li>Applied Bayesian statistics</li>
<li>Advance data visualization</li>
<li>Randomized experiments</li>
<li>Quasi-experimental methods</li>
<li>Programming in Python or R</li>
<li>Advance databases</li>
<li>Cloud and distributed computing</li>
</ul>
<h3>Branches</h3>
<p>Finally, the branches of the tree represent topics that require intense dedication and expertise. Think of this as acquiring a medical specialty, like pediatrician or internal medicine. This means I do not believe everyone should learn about these topics. Actually, as many physicians do, you may skip this section altogether and become a perfectly high-quality and competent professional by mastering some cluster of topics from the trunk of the tree. When I see people offering courses to master stuff like &ldquo;Computer vision&rdquo; or &ldquo;Graphical models&rdquo; in a 2 months course, I cannot avoid but to feel such a person is just trying to profit from the hopes and dreams of hardworking people. True, there are examples of people that learned a lot from such courses, and some went as far as getting a job in that specialty. But those are exceptions, not the rule. I do not have the numbers here, but I am pretty sure they are not many computer vision experts who learned their craft in a 2 months Udemy course. By the way, I do not look down at those courses at all! I have taken (yes, I PAID) $11 Udemy courses as well. There are fine but they are no replacement for proper Data Science education. Notice I say &ldquo;education&rdquo; not &ldquo;degree&rdquo;, which I think are different things: self-educated people is valuable and competent as any fancy degree-holder, but self-education ought to be done seriously, meaning a plan, meaning long hours, meaning years of effort. In any case, these are topics that require two things: a solid foundation in general Data Science topics (see now?), and significant time and effort as their complexity requires:</p>
<ul>
<li>Time-series analysis</li>
<li>Advance applied statistics</li>
<li>Computer vision</li>
<li>Natural language processing</li>
<li>Reinforcement learning</li>
<li>Deep reinforcement learning</li>
<li>Graphical models</li>
<li>Advance artificial intelligence</li>
<li>High-performance computing</li>
<li>High-throughput computing</li>
<li>Cloud computing</li>
<li>Representation learning</li>
<li>Deep generative models</li>
<li>Advance Bayesian models</li>
<li>Monte-carlo methods</li>
<li>Affective computing</li>
<li>Human-computer interaction</li>
<li>Robotics</li>
<li>Hybrid models</li>
</ul>
<p>My roots, trunk, and branches lists are by no means complete or exhaustive, but I hope are enough to convey my main points: <strong>Data Science, as a discipline, does require a sensible core of knowledge to define itself as a discipline, and to envision a clear learning path for people who aspire to become a Data Scientist</strong>.</p>
</article>
</section>
</div>
</div>
<br>
<br>
<br>
<br>
<br>
<br>
via pabloinsente.github.io https://ift.tt/2USUrGD<br>
<br>
November 24, 2020 at 03:06PM
